src/test/java/dev/sigstore/encryption/KeysTest.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2022 The Sigstore Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package dev.sigstore.encryption;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import com.google.common.io.Resources;
+import java.io.IOException;
+import java.security.NoSuchAlgorithmException;
+import java.security.PublicKey;
+import java.security.spec.InvalidKeySpecException;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+class KeysTest {
+
+  static final String RSA_PUB_PATH = "dev/sigstore/samples/keys/test-rsa.pub";
+  static final String EC_PUB_PATH = "dev/sigstore/samples/keys/test-ec.pub";
+  static final String ED25519_PUB_PATH = "dev/sigstore/samples/keys/test-ed25519.pub";
+  static final String DSA_PUB_PATH = "dev/sigstore/samples/keys/test-dsa.pub";
+
+  @Test
+  void parsePublicKey_rsa() throws IOException, InvalidKeySpecException, NoSuchAlgorithmException {
+    PublicKey result =
+        Keys.parsePublicKey(Resources.toByteArray(Resources.getResource(RSA_PUB_PATH)));
+    assertEquals(result.getAlgorithm(), "RSA");
+  }
+
+  @Test
+  void parsePublicKey_ec() throws IOException, InvalidKeySpecException, NoSuchAlgorithmException {
+    PublicKey result =
+        Keys.parsePublicKey(Resources.toByteArray(Resources.getResource(EC_PUB_PATH)));
+    assertEquals(result.getAlgorithm(), "EC");
+  }
+
+  @Test
+  void parsePublicKey_ed25519()
```

Comment:
I believe the "proper" way to do this woud be to write two tests, one for each side of jdk15, and use the annotation [EnabledForJreRange](https://junit.org/junit5/docs/5.8.1/api/org.junit.jupiter.api/org/junit/jupiter/api/condition/EnabledForJreRange.html)

kinda like
```
@EnabledForJreRange(max = JRE.JAVA_14)
void parsePublicKey_ed25519_withBouncyCastle() {
```

and 
```
@EnabledForJreRange(min = JRE.JAVA_15)
void parsePublicKey_ed25519_withStdLib() {
```

Reply:
Nice tip!!  Switched to those.